pkg/skaffold/build/tag/custom.go

```diff
@@ -27,18 +27,28 @@ type CustomTag struct {
 	Tag string
 }
 
-func (c *CustomTag) Labels() map[string]string {
+// Labels are labels specific to the custom tagger.
+func (t *CustomTag) Labels() map[string]string {
 	return map[string]string{
 		constants.Labels.TagPolicy: "custom",
 	}
 }
 
-// GenerateFullyQualifiedImageName tags an image with the custom tag
-func (c *CustomTag) GenerateFullyQualifiedImageName(workingDir, imageName string) (string, error) {
-	tag := c.Tag
+// GenerateTag resolves the tag portion of the fully qualified image name for an artifact.
+func (t *CustomTag) GenerateTag(workingDir, imageName string) (string, error) {
+	tag := t.Tag
 	if tag == "" {
 		return "", errors.New("custom tag not provided")
 	}
+	return tag, nil
+}
+
+// GenerateFullyQualifiedImageName tags an image with the custom tag.
+func (t *CustomTag) GenerateFullyQualifiedImageName(workingDir, imageName string) (string, error) {
```

Comment:
Maybe worth creating a common function for most of the implementations to call?  (except the envTemplateTagger)

Reply:
I will be creating a common function for all implementations to call. We will be deprecating envTemplateTagger's use of {{.IMAGE_NAME}} and assume the template only specifies the tag value.